variables.tf

```diff
@@ -1,11 +1,154 @@
-##############################################################################
-# Input Variables
-##############################################################################
+variable "resource_group_id" {
+  description = "ID of resource group to use when creating the event stream instance"
+  type        = string
+}
 
-#variable "myvariable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = ""
-#}
+variable "es_name" {
+  type        = string
+  description = "Name of the resource instance"
+}
 
-##############################################################################
+variable "plan" {
+  type        = string
+  description = "Plan for the event streams instance : lite, standard or enterprise-3nodes-2tb"
+  default     = null
+  validation {
+    condition     = contains(["lite", "standard", "enterprise-3nodes-2tb"], var.plan)
+    error_message = "The specified plan is not a valid selection! Supported plans are: lite, standard or enterprise-3nodes-2tb."
+  }
+}
+
+variable "tags" {
+  type        = list(string)
+  description = "List of tags associated with the Event Steams instance"
+  default     = []
+}
+
+variable "region" {
+  type        = string
+  description = "IBM Cloud region where event streams will be created"
+  default     = "us-south"
+}
+
+variable "throughput" {
+  type        = number
+  description = "Throughput capacity in MB per second."
+  default     = "150" # For lite and standard plan, the allowed value of throughput is 150 MB per second. For enterprise plan it can take any value.
+}
+
+variable "storage_size" {
+  type        = number
+  description = "Storage size of the event streams in GB."
+  # For lite and standard plan, the allowed value is 2048 GB. For enterprise plan refer (https://cloud.ibm.com/docs/EventStreams?topic=EventStreams-ES_scaling_capacity#ES_storage_capacity) for allowed values.
+  # Storage capacity cannot be scaled down once instance is created.
+  # Storage capacity cannot be scaled up in lite and standard plan.
+  default = null
```

Comment:
This should probably not have a default value of `null`. What happens if null is passed? TF prob fails? Based on the code comment (which should be moved into the variable description), it seems `2048` would be the right default value?

Reply:
A lot of these variables seem to apply to only enterprise instances https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/event_streams_schema.